msal/src/main/java/com/microsoft/identity/client/MultipleAccountPublicClientApplication.java

```diff
@@ -379,38 +353,31 @@ private void removeAccountInternal(@Nullable final IAccount account,
                         requestAccountRecord
                 );
 
-        try {
-            final RemoveAccountCommand removeAccountCommand = new RemoveAccountCommand(
-                    params,
-                    MSALControllerFactory.getAllControllers(
-                            mPublicClientConfiguration.getAppContext(),
-                            mPublicClientConfiguration.getDefaultAuthority(),
-                            mPublicClientConfiguration
-                    ),
-                    new CommandCallback<Boolean, BaseException>() {
-                        @Override
-                        public void onError(BaseException error) {
-                            callback.onError(MsalExceptionAdapter.msalExceptionFromBaseException(error));
-                        }
-
-                        @Override
-                        public void onTaskCompleted(Boolean success) {
-                            callback.onRemoved();
-                        }
-
-                        @Override
-                        public void onCancel() {
-                            //Do nothing
-                        }
-                    },
-                    publicApiId
-            );
+        final RemoveAccountCommand removeAccountCommand = new RemoveAccountCommand(
+                params,
+                new MSALControllerFactory(mPublicClientConfiguration).getAllControllers(
+                        mPublicClientConfiguration.getDefaultAuthority()
+                ),
+                new CommandCallback<Boolean, BaseException>() {
+                    @Override
+                    public void onError(BaseException error) {
+                        callback.onError(MsalExceptionAdapter.msalExceptionFromBaseException(error));
+                    }
 
-            CommandDispatcher.submitSilent(removeAccountCommand);
+                    @Override
+                    public void onTaskCompleted(Boolean success) {
+                        callback.onRemoved();
+                    }
 
-        } catch (final MsalClientException e) {
-            callback.onError(e);
-        }
```

Comment:
We no longer need to catch this?

Reply:
yes - since we no longer throws exception from MSALControllerFactory. 

If MSALFactory hits any AccountManager error it will simply return LocalMsalController. (see LegacyBrokerDiscoveryClient.getActiveBroker())

The exception handler of the actual operation is passed via callback. no change there.